### ISSUE TITLE
Using openlayers 4.6.5 to add MVT in Vue components causes a lot of memory leaks

### ISSUE THREAD

**Issue by a24576946** [openlayers/openlayers#9531]
This problem does not occur in normal pages. Memory leaks are very serious in the components of vue. Even if we just initialize the general background map, memory leaks gradually and can not be released.

**Comment by ahocevar** [comment]
Can you please update to the latest version (currently v6.0.0-beta.7)? These issues have been fixed in the meantime.

**Comment by fredj** [comment]
@a24576946 is this issue related to #9556 ?

**Comment by a24576946** [comment]
@ahocevar @fredj Thank you for your answer, but how do I install version 6.0.0? It seems that I can only install version 4.6.5 through npm?

**Comment by fredj** [comment]
The latest beta version is available on npm, see: https://www.npmjs.com/package/ol/v/6.0.0-beta.9

**Comment by a24576946** [comment]
@fredj 
Thank you very much, but I have used a lot of the way to instantiate ol objects like Code 1 in my project. The version you provided to me seems to only support the way to instantiate ol objects like Code 2, but not the way I use it now.
Code1：`let wktFormat = new ol.format.WKT();`
Code2：`import Map from 'ol/Map';import View from 'ol/View';`
Is there a version of this link instantiating the ol object?

**Comment by a24576946** [comment]
@fredj Thank you. What I want is a way to instantiate ol objects like Code1.

**Comment by a24576946** [comment]
@fredj @ahocevar I have used the latest version of ol, but when loading mvt, there will still be a lot of memory increase problems, and only in the Vue components, the introduction of openlayers in normal pages will not occur this problem, how to solve this problem?
![load mvt](https://user-images.githubusercontent.com/35019456/59410403-57b7b900-8deb-11e9-929d-32f3763e6c9a.png)
![problem](https://user-images.githubusercontent.com/35019456/59410418-5e463080-8deb-11e9-9d4a-8dc6780712b9.png)
![version](https://user-images.githubusercontent.com/35019456/59410422-63a37b00-8deb-11e9-94dc-e9ea162c760f.png)

**Comment by ahocevar** [comment]
In the meantime we fixed another memory leak, see #9637. That's not in a beta release yet though.

**Comment by a24576946** [comment]
@ahocevar @fredj 
Thank you for your help and advice. :grinning:You can visit this page to see my memory leak example.
https://a24576946.github.io/memoryLeakPage/#/
This is the memory leak display page after packaging. You can also visit this URL to see the code I wrote.
https://github.com/a24576946/memoryLeakCode

**Comment by a24576946** [comment]
@ahocevar @fredj 
There is also a bug in openlayers6.0.0-beta9. If two layers are added, the first layer is the basemap service of the image wmts, the second layer is the mvt layer of the point type, and the two Layers are set to the zindex level parameter, there will be a bug like the picture, and the generated mvt dot can not be cleared, but the bug does not exist in the line type and the polygon type mvt.
![1](https://user-images.githubusercontent.com/35019456/59488356-f4448e80-8eb1-11e9-811c-4948511f32df.png)
![2](https://user-images.githubusercontent.com/35019456/59488373-f9094280-8eb1-11e9-8b4b-ffb0ccf73d3d.png)

**Comment by ahocevar** [comment]
@a24576946 This has been fixed in the meantime. The fix will be in beta 10.

**Comment by a24576946** [comment]
@ahocevar @fredj Thank you for your hard work. We have thought of some ways to avoid this bug. The previous problem of memory leak in MVT is caused by the closure environment we call in Vue components. It is not particularly relevant to your openlayers.

**Comment by a24576946** [comment]
@ahocevar @fredj In addition, we continue to make the point type can not appear in the reappearance, how soon will beta 10 be updated?

**Comment by ahocevar** [comment]
@a24576946 I'm planning to release v6.0.0-beta.10 today.

**Comment by a24576946** [comment]
`      this.mapTools.map.once("rendercomplete", function(event) {
        let canvas = event.context.canvas;
        let base64 = canvas.toDataURL("image/jpeg", 0.5);
        This.coverImg = base64;
      });`
@ahocevar Hello, as shown in the figure, we found a bug in the process of testing the new version. In the event object of the event 'rendercomplete' of 6.0.0-beta9, the value of the context property is undefined, but in the In versions prior to 5.3.0, the event object of the 'rendercomplete' event can get the value of the context property.
![canvasbug](https://user-images.githubusercontent.com/35019456/59589346-4983d880-911c-11e9-8983-0fce2d74c134.png)

**Comment by ahocevar** [comment]
@a24576946 That has also been fixed in the meantime.

**Comment by a24576946** [comment]
@ahocevar Thank you for your hard work and update, the mvt memory leak problem has been solved to a large extent, but we also found a new problem, as shown in the figure, if I load the mvt layer of polygon type, it will appear part The layer is missing, and my layer request is successful. This problem only exists in the polygon type. There is no such bug in line and point.
![1](https://user-images.githubusercontent.com/35019456/60075087-17dfc280-9757-11e9-9b06-794e6e155f5c.png)
